backoff_test.go

```diff
@@ -9,3 +9,32 @@ func TestBackoffConfigDefaults(t *testing.T) {
 		t.Fatalf("expected BackoffConfig to pickup default parameters: %v != %v", b, DefaultBackoffConfig)
 	}
 }
+
+func TestBackoffWitDifferentNumberOfRetries(t *testing.T) {
+	const maxRetries = 10
+	for retries := 0; retries < maxRetries; retries++ {
+		b := BackoffConfig{}
+		setDefaults(&b)
+		backoffTime := b.backoff(retries)
+		// Backoff time should be between basedelay and max delay.
+		if backoffTime < b.baseDelay || backoffTime > b.MaxDelay {
+			t.Fatalf("expected backoff time: %v to be between basedelay: %v and maxdelay: %v", backoffTime, b.baseDelay, b.MaxDelay)
+		}
+	}
+}
+
+func TestBackOffTimeIncreasesWithRetries(t *testing.T) {
+	const maxRetries = 10
+	b := BackoffConfig{}
+	setDefaults(&b)
+	// Base delay.
+	lastBackOffTime := b.backoff(0)
+	for retries := 1; retries <= maxRetries; retries++ {
```

Comment:
some nit:
change "<=" to "<" so that "maxRetries" semantics is consistent across these 2 tests?